### PR TITLE
Prepend select fields

### PR DESCRIPTION
**Pull request by Gberzi01**
1. Prepend all select field type fields with empty values
2. Fix select field to not work as a relation but only as a model field value picker
3. Allow saving select with empty values
4. Add defaultValue to all fields, it is up to the developer to pass correct value types
    - For example, to check all belongsToMany fields one should use Model::all(), but to select only a single field collect([Model::first()])
    - For most fields string works
    - Checkbox works with 1/0 or true/false

### CODE REVIEW

File: src/Admin/Form/Fields/AbstractField.php [hunk 1]
```diff
@@ -126,9 +132,18 @@ public function getValue()
...
         }
 
+        if (($this->value === null || $this->isEmptyCollection($this->value)) && $this->getDefaultValue()) {
+            $this->value = $this->getDefaultValue();
+        }
+
         return $this->value;
     }
 
+    private function isEmptyCollection($value)
```
Comment (holystix) [inline src/Admin/Form/Fields/AbstractField.php]: Missing docblock.

File: src/Admin/Form/Fields/AbstractField.php [hunk 2]
```diff
@@ -126,9 +132,18 @@ public function getValue()
             $this->value = $this->getModel()->getAttribute( $this->getName() );
         }
 
+        if (($this->value === null || $this->isEmptyCollection($this->value)) && $this->getDefaultValue()) {
```
Comment (holystix) [inline src/Admin/Form/Fields/AbstractField.php]: `($this->value === null || $this->isEmptyCollection($this->value)` part of the condition should be ported to a separate function for better readability.

File: src/Admin/Form/Fields/AbstractField.php [hunk 3]
```diff
@@ -140,6 +155,23 @@ public function setValue( $value )
         return $this;
     }
 
+    /** @return mixed */
```
Comment (holystix) [inline src/Admin/Form/Fields/AbstractField.php]: Should be multiline docblock for consistency.

File: src/Admin/Form/Fields/Concerns/HasSelectOptions.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+use Illuminate\Support\Collection;
+
+/**
+ * Class HasSelectOptions
+ * @package Arbory\Base\Admin\Form\Fields\Concerns
+ */
+trait HasSelectOptions
+{
+    /** @var Collection $options */
```
Comment (holystix) [inline src/Admin/Form/Fields/Concerns/HasSelectOptions.php]: Docblock should be multiline.

File: src/Admin/Form/Fields/AbstractField.php (hunk 1 above)
Comment (Gberzi01) [reply src/Admin/Form/Fields/AbstractField.php]: Fixed in 9f4560d

**Pull request merged**